packages/neuron-wallet/src/services/indexer/transaction-cache.ts

```diff
@@ -0,0 +1,36 @@
+import { TransactionWithStatus } from "types/cell-types"
+
+export class TransactionCache {
+  private limit: number
+
+  private store = new Map<string, TransactionWithStatus>()
+
+  constructor(limit: number) {
+    this.limit = limit
+  }
+
+  public push(value: TransactionWithStatus) {
+    const key = value.transaction.hash
+
+    if (this.store.has(key)) {
+      return
+    }
+    this.store.set(key, value)
+    if (this.store.size > this.limit) {
+      const firstKey = Array.from(this.store.keys())[0]
```

Comment:
You can get the first key by `this.store.keys().next().value`

Reply:
OK, I mistakenly thought the `value` is the value of a `Map`, not the key .... 😢